### PR TITLE
Fix #3362 Use OppiaTestRule in all Espresso Tests 

### LINKED ISSUE #3362: Use `OppiaTestRule ` in all Espresso Tests
Use `OppiaTestRule` in all Espresso Tests

Basically every Espresso test file by default should inject this rule.

### PR DESCRIPTION
**Pull request by vrajdesai78**
## Explanation
Fix #3362 by fixing all to-dos which were mentioned by Github-Actions. I have removed AccessibilityChecks according to mentioned To-dos. 

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## Screenshots of passing testcases
Screenshot from 2022-02-21 18-41-14
Screenshot from 2022-02-21 18-47-27
Screenshot from 2022-02-21 18-48-59
Screenshot from 2022-02-21 22-37-20
Screenshot from 2022-02-21 22-38-19
Screenshot from 2022-02-21 22-59-39
Screenshot from 2022-02-22 00-02-26

## Issues Filed for failing test cases
- #4212 
- #4149 
- #4209

### CODE REVIEW

File: app/src/sharedTest/java/org/oppia/android/app/story/StoryActivityTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/story/StoryActivityTest.kt]: If you removing `DisableAccessibilityChecks ` it means that the A11Y check is enabled and therefore TODO is not required.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@vrajdesai78 Three suggestions:

1. If you have removed `DisableAccessibilityChecks` then `// TODO(#3362): Enable AccessibilityChecks` is not required.
2. If you have added `DisableAccessibilityChecks` please add a `comment+TODO` mentioning why this is needed, maybe explain the type of error.
3. For all test files which are changed in this PR please add screenshot of them running on Espresso.

File: app/src/sharedTest/java/org/oppia/android/app/story/StoryActivityTest.kt
Comment (vrajdesai78) [reply app/src/sharedTest/java/org/oppia/android/app/story/StoryActivityTest.kt]: What issue number should I add in TODO can I add #3245 as it is related to enabling Accessibility Checks or current issue number if I add DisableAccessbilityChecks

Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/android/app/story/StoryActivityTest.kt]: Use #3245 for all `DisableAccessbilityCheckes ` annotations

**Comment by vrajdesai78** [comment]
> What issue number should I add in TODO can I add #3245 as it is related to enabling Accessibility Checks or current issue number if I add DisableAccessbilityCheckes

@rt4914 PTAL.

**Comment by vrajdesai78** [comment]
Passing Testcases after DisablingAccessbility Checks
Screenshot from 2022-02-14 16-52-05
Screenshot from 2022-02-14 16-52-56
Screenshot from 2022-02-14 16-53-39

File: app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt]: `// TODO(#3245): Throw error URLSpan should be used in place of ClickableSpan`

`Error` -> `error`

File: app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt]: `Error ` -> `error`

File: app/src/sharedTest/java/org/oppia/android/app/databinding/ImageViewBindingAdaptersTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/databinding/ImageViewBindingAdaptersTest.kt]: Please add screenshot of espresso tests passing for this file too.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@vrajdesai78 I think the above suggestions are not fully applied yet.

1. There are many new files which uses `OppiaTestRule` in this PR and I am unable to see screenshots of espresso-test for all of them.
2. In `PinPasswordActivityTest` uses `DisableAccessibilityChecks ` and there are no comments for it.
3. If there is any `...TestActivityTest` like `InputInteractionViewTestActivityTest `. In this along with `DisableAccessibilityChecks` please add one comment (not TODO) mentioning `Disabled as this is a test file and will not be used by user.` (We might update this comment later but for now having a comment is important)

File: app/src/sharedTest/java/org/oppia/android/app/databinding/ImageViewBindingAdaptersTest.kt
Comment (vrajdesai78) [reply app/src/sharedTest/java/org/oppia/android/app/databinding/ImageViewBindingAdaptersTest.kt]: Actually, there is an issue with this Test. I have already created an issue for same #4149

**Comment by vrajdesai78** [comment]
> @vrajdesai78 I think the above suggestions are not fully applied yet.
> 
> 1. There are many new files which uses `OppiaTestRule` in this PR and I am unable to see screenshots of espresso-test for all of them.
> 2. In `PinPasswordActivityTest` uses `DisableAccessibilityChecks ` and there are no comments for it.
> 3. If there is any `...TestActivityTest` like `InputInteractionViewTestActivityTest `. In this along with `DisableAccessibilityChecks` please add one comment (not TODO) mentioning `Disabled as this is a test file and will not be used by user.` (We might update this comment later but for now having a comment is important)

Actually, I have added all the screenshots in #4133.

**Comment by vrajdesai78** [comment]
@rt4914 PTAL.

File: app/src/sharedTest/java/org/oppia/android/app/profile/PinPasswordActivityTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/profile/PinPasswordActivityTest.kt]: @BenHenning Have a look at this comment and let us know if you think it clarifies everything.

File: app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityTest.kt]: This is not a test file. This is test file for `WalkthroughActivity` so please add correct TODO. In this entire file

**Review by rt4914 — CHANGES_REQUESTED** [review]
@vrajdesai78 Suggested changes. Also you can copy the images from your other PR by clicking on "Edit" and then paste it in PR description. Your PR description should contain all the information about tests are passing or not, why a particular change was done, etc.

**Comment by vrajdesai78** [comment]
I have already created issues #4148 and #4149 for ImageViewBindingAdaptersTest and PinPasswordActivityTest. @rt4914 PTAL. Thanks

margin_bind
Screenshot from 2022-01-28 19-12-28
Screenshot from 2022-01-28 19-14-25
Screenshot from 2022-01-28 19-16-05
Screenshot from 2022-01-28 20-08-42
Screenshot from 2022-01-28 20-12-11
Screenshot from 2022-01-28 20-13-20
Screenshot from 2022-01-28 20-33-28
Screenshot from 2022-01-28 20-34-27
Screenshot from 2022-01-28 20-35-19
Screenshot from 2022-01-28 20-36-33
Screenshot from 2022-01-28 22-25-24
Screenshot from 2022-01-28 22-33-27
Screenshot from 2022-01-28 22-34-11
stateAssembler
Screenshot from 2022-01-28 22-30-36

File: app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityTest.kt
Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityTest.kt]: @vrajdesai78 This is pending

File: app/src/sharedTest/java/org/oppia/android/app/profile/PinPasswordActivityTest.kt
Comment (rt4914) [reply app/src/sharedTest/java/org/oppia/android/app/profile/PinPasswordActivityTest.kt]: If if you have filed the issue then please mention that issue number as TODO.

**Review by rt4914 — CHANGES_REQUESTED** [review]
Please make following changes to your PR description.

1. Screenshots of tests passing. (Shift them from comments to description)
2. Issues filed. (Shift them from comments to description)
3. Also if you have used `DisableAccessibibilityChecks` then please mention TODO in comments corresponding to correct issue number and if you dont have any issue number, then just file a new issue accordingly.

File: app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt:216]: This test has a TODO but has not been disabled?

File: app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt:231]: This test has a TODO but has not been disabled?

File: app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt]: Please mention it as a single TODO with both the errors. 

`// TODO(#3245): Use URLSpan and minimum touch target size should be 48x48 dp`

File: app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt]: Reduce to single line and don't mention ids. Just the error info is enough.

File: app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt]: Ditto

File: app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt]: Ditto

**Review by rt4914 — CHANGES_REQUESTED** [review]
@vrajdesai78 Suggested changes.

File: app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt
Comment (vrajdesai78) [reply app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt:216]: Actually, this issue is not related to AccessbilityChecks that's why I haven't disable Accessibility checks. This TODO is related to #4212 as mentioned in the PR description.

File: app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt]: @vrajdesai78 Please make these consistent. Sometimes you have mentioned TODO on top of `@Test`, sometimes at bottom and sometimes in between @Test and `@Config`

I suggest writing TODO next to @Test of at the top of @Config and @Test

**Review by rt4914 — APPROVED** [review]
@vrajdesai78 LGTM. Suggested one minor change. When writing such code please focuse on consistency.

File: app/src/sharedTest/java/org/oppia/android/app/testing/InputInteractionViewTestActivityTest.kt
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/testing/InputInteractionViewTestActivityTest.kt]: Can we specify the class name rather than using `this`?
Example `// Disabled, as InputInteractionViewTestActivity is a test file and will not be used by user`

**Review by anandwana001 — APPROVED** [review]
LGTM,

left a nit suggestion.

File: app/src/sharedTest/java/org/oppia/android/app/testing/InputInteractionViewTestActivityTest.kt
Comment (vrajdesai78) [reply app/src/sharedTest/java/org/oppia/android/app/testing/InputInteractionViewTestActivityTest.kt]: @anandwana001  Done

File: app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt
Comment (vrajdesai78) [reply app/src/sharedTest/java/org/oppia/android/app/story/StoryFragmentTest.kt]: @rt4914 Done

**Comment by BenHenning** [comment]
It appears my codeowners is no longer needed. I think @rt4914 just needs to verify the remaining changes before this can be merged.

**Pull request merged**
